src/dotnet/Chat.UI.Blazor/Components/ChatList/ChatListItem.razor

```diff
@@ -1,101 +1,147 @@
+                            <i class="icon-more-vertical text-2xl"></i>
+                        </ButtonRound>
+                    </div>
                 }
-                <div class="@(hasMenu ? "" : "pr-1")">
-                    <ChatListListenToggle AudioState="@audioState" Class="blackout"/>
-                </div>
-            }
-            @if (hasMenu) {
-                <div
-                    class="menu"
-                    data-menu="@(MenuRef.New<ChatMenu>(chat.Id, ListKind.Format()).ToString())"
-                    data-menu-trigger="@MenuTrigger.Primary"
-                    data-menu-placement="@(FloatingPosition.RightStart.ToPositionString())">
-
-                    <ButtonRound Class="blackout" ContentClass="!bg-transparent">
-                        <i class="icon-more-vertical text-2xl"></i>
-                    </ButtonRound>
-                </div>
-            }
-        </div>
-    </Ending>
-</NavbarItem>
+            </div>
+        </Ending>
+    </NavbarItem>
+    @if (isLastPinned) {
+        <div class="mx-2 my-1.5 border-b border-bg-04"></div>
+    }
+}
 
 @code {
-    private ChatState? _rendered;
+    private Model? _rendered;
 
     [Inject] private ChatUI ChatUI { get; init; } = null!;
+    [Inject] private SearchUI SearchUI { get; init; } = null!;
 
     [CascadingParameter] public ScreenSize ScreenSize { get; init; }
-    [Parameter, EditorRequired] public ChatInfo ChatInfo { get; init; } = null!;
-    [Parameter, EditorRequired] public ChatListKind ListKind { get; init; }
+
+    // Provide Position for ChatListKind.All or ChatInfo for ChatListKind.Active
+    [Parameter] public int Position { get; init; }
+    [Parameter] public ChatInfo ChatInfo { get; init; } = null!;
+
+    [Parameter] public ChatListKind ListKind { get; init; } = ChatListKind.All;
 
     protected override bool ShouldRender()
         => State.HasError || !ReferenceEquals(_rendered, State.Value);
 
-    protected override ComputedState<ChatState>.Options GetStateOptions()
+    protected override ComputedState<Model>.Options GetStateOptions()
         => new() {
-            InitialValue = new ChatState(ChatInfo, new ChatAudioState(ChatInfo.Id)),
+            InitialValue = Model.Loading,
             Category = GetStateCategory(),
         };
 
-    protected override async Task<ChatState> ComputeState(CancellationToken cancellationToken)
-    {
-        var chatId = await ChatUI.FixChatId(ChatInfo.Id, cancellationToken).ConfigureAwait(false);
-        var chatState = await ChatUI.GetState(chatId, false, cancellationToken).ConfigureAwait(false);
-        return chatState ?? ChatState.None;
+    protected override async Task<Model> ComputeState(CancellationToken cancellationToken) {
+        ChatInfo? chatInfo;
+        if (ListKind == ChatListKind.Active)
+            chatInfo = ChatInfo;
+        else {
+            var chatId = await ChatUI.GetChatId(Position, cancellationToken).ConfigureAwait(false);
```

Comment:
afair we agreed not to use ConfigureAwait(false) in components

Reply:
Yep.